code-challenges.test.js

```diff
@@ -23,10 +24,31 @@ var secretCodeWord2 = "Gobbledygook"
 var secretCodeWord3 = "Eccentric"
 // Expected output: "3cc3ntr1c"
 
+describe("codedMessage", () => {
+    it("that takes in a string and returns a coded message", () => {
+        expect(codedMessage(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+        expect(codedMessage(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+        expect(codedMessage(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+})
 
+// ReferenceError: codedMessage is not defined
 
 // b) Create the function that makes the test pass.
 
+const codedMessage = (string) => {
+    // Create a function called codedMessage  that takes in a string as a parameter 
+    finalString = ''
+    // Create an empty string to contain the new string since .replace does not alter the original string
+    for(let i=0; i < string.length; i++){
+        // Iterate through the string with a for loop
+        finalString = string.replace(/([aA])/g, "4").replace(/([eE])/g, '3').replace(/([iI])/g, '1').replace(/[oO]/g, '0')
+        }
+        // Use the .replace method with regex to replace each vowel with the corresponding number. Perform a global replacement to replace all instances of the vowel.
+        return finalString
+    }
+
```

Comment:
Alright this works! But it doesn't need a for loop. For loops give you access to the value if "i". 

You aren't calling on "i" and you have a return in your for loop that stops the for loop. So here is your code just refactored to look like what it is already doing.
```javascript 
// Create a function called codedMessage  that takes in a string as a parameter
const codedMessage = (string) => {
    // Create an empty string to contain the new string since .replace does not alter the original string
    finalString = ''
        // Use the .replace method with regex to replace each vowel with the corresponding number. Perform a global replacement to replace all instances of the vowel.
   finalString = string.replace(/([aA])/g, "4").replace(/([eE])/g, '3').replace(/([iI])/g, '1').replace(/[oO]/g, '0')
        
      
   return finalString
}
```